### PR TITLE
PPS capture driver (not PPS clock synchronization)

### PR DESCRIPTION
**Pull request by Igor-Misic**
PPS - pulse per second 
RTC - real-time clock
EXTI - external interrupt
HRT - high-resolution timer

This is a driver that uses PPS from GNSS module to calculate time correction separately from the RTC. It uses GPIO EXTI as it is intended to be used at boards where is not possible to connect PPS to a pin that can capture HRT timestamp. 
For boards that have an RTC clock synchronized with PPS, this driver is not needed. 

This PR also brings improvements for camera capture and camera trigger.

### CODE REVIEW

File: src/drivers/pps_capture/PPSCapture.hpp [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+	/** PPSCapture is an interrupt-driven task and needs to be manually stopped */
+	static void stop();
+
+private:
+	void Run() override;
+
+	int _channel{-1};
+	uint32_t _pps_capture_gpio{0};
+	uORB::Publication<pps_capture_s>	_pps_capture_pub{ORB_ID(pps_capture)};
+	uORB::Subscription								_sensor_gps_sub{ORB_ID(sensor_gps)};
```
Comment (dagar) [inline src/drivers/pps_capture/PPSCapture.hpp:75]: Having multiple GPS units is becoming common.

File: src/drivers/pps_capture/PPSCapture.cpp [hunk 2]
```diff
@@ -0,0 +1,221 @@
...
+ * This is driver for capturing GNSS Pulse Per Second (PPS) signal.
+ *
+ */
+
+#include "PPSCapture.hpp"
+#include <px4_arch/io_timer.h>
+#include <board_config.h>
+#include <parameters/param.h>
+
+#ifdef BOARD_WITH_IO
```
Comment (bkueng) [inline src/drivers/pps_capture/PPSCapture.cpp:46]: Can you apply https://github.com/PX4/PX4-Autopilot/pull/18860?

File: src/modules/logger/logged_topics.cpp [hunk 3]
```diff
@@ -172,6 +172,7 @@ void LoggedTopics::add_default_topics()
 	add_optional_topic_multi("sensor_accel", 1000, 4);
 	add_optional_topic_multi("sensor_baro", 1000, 4);
 	add_topic_multi("sensor_gps", 1000, 2);
+	add_topic("pps_capture", 1000);
```
Comment (bkueng) [inline src/modules/logger/logged_topics.cpp:175]:
```suggestion
	add_optional_topic("pps_capture", 1000);
```

File: src/drivers/pps_capture/PPSCapture.cpp [hunk 4]
```diff
@@ -0,0 +1,221 @@
...
+# define PARAM_PREFIX "PWM_AUX"
+#else
+# define PARAM_PREFIX "PWM_MAIN"
+#endif
+
+PPSCapture::PPSCapture() :
+	ScheduledWorkItem(MODULE_NAME, px4::wq_configurations::hp_default)
+{
```
Comment (bkueng) [inline src/drivers/pps_capture/PPSCapture.cpp:54]: `_pps_capture_pub.advertise();` (for logging)

**Review by bkueng — APPROVED** [review]

**Pull request merged**